File: 🟡 Medium/2285 - Maximum Total Importance of Roads/_2285MaxTotalImpOfRoads.js
var maximumImportance = function (n, roads) {
    let freq = [];
    for (let i = 0; i < n; i++) {
        freq[i] = 0;
    }

    for (const edge of roads) {
        freq[edge[0]]++;
        freq[edge[1]]++;
    }

    freq.sort((a, b) => a - b);

    let imp = 0;
    let k = 1;

    for (const f of freq) {
        imp += (k * f);
        k++;
    }

    return imp;
};

File: 🟡 Medium/2285 - Maximum Total Importance of Roads/_2285MaxTotalImpOfRoads.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<long long> freq(n);

        for (vector<int>& edge : roads) {
            freq[edge[0]]++;
            freq[edge[1]]++;
        }

        sort(freq.begin(), freq.end());

        long imp = 0;
        long k = 1;

        for (long f : freq) {
            imp += (k * f);
            k++;
        }

        return imp;
    }
};